include/dataformats/wib2/WIB2Unpack.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ *
+ *  @file WIB2Unpack.hpp
+ *
+ *  Functions for manipulating WIB2 frames in the format described by
+ *  EDMS document 2088713:
+ *  (https://edms.cern.ch/document/2088713/4). The code here is taken
+ *  from unpack.h in the dune-wib-firmware repository, written by Ben
+ *  Land, and lightly modified to be closer to DUNE DAQ software
+ *  conventions. It is used in this repository as a check of the
+ *  overlay code in WIB2Frame.hpp
+ *
+ */
+#ifndef DAQRAWDATA_WIB2UNPACK_HPP
+#define DAQRAWDATA_WIB2UNPACK_HPP
+
+#include <cstddef>
+#include <stdint.h>
+#include <vector>
+
+namespace dunedaq::dataformats::wib2 {
+
+// Words in the binary format of the FELIX frame14 from the WIB
+typedef struct
+{
+  uint32_t start_frame;
+  uint32_t wib_pre[4];
+  uint32_t femb_a_seg[56];
+  uint32_t femb_b_seg[56];
+  uint32_t wib_post[2];
+  uint32_t idle_frame;
+} __attribute__((packed)) frame14;
+
+// Samples from the U, V, X channels in a femb_*_seg of a frame as 16bit arrays
+typedef struct
+{
+  uint16_t u[40], v[40], x[48];
+} __attribute__((packed)) femb_data;
+
+// Byte-aligned unpacked values in a felix_frame
+typedef struct
+{
+  uint8_t link_mask, femb_valid, fiber_num, wib_num, frame_version, crate_num;
+  uint32_t wib_data;
+  uint64_t timestamp;
+  femb_data femb[2];
+  uint32_t crc20;
+  uint16_t flex12;
+  uint32_t flex24;
+} frame14_unpacked;
+
+// Deframed data, where channels or u,v,x are time ordered uint16 samples for each channel
+typedef struct
+{
+  size_t samples;
+  std::vector<uint16_t> channels[2][128];
+  std::vector<uint64_t> timestamp;
+  uint8_t crate_num, wib_num;
+} channel_data;
+
+typedef struct
+{
+  size_t samples;
+  std::vector<uint16_t> u[2][40];
+  std::vector<uint16_t> v[2][40];
+  std::vector<uint16_t> x[2][48];
+  std::vector<uint64_t> timestamp;
+  uint8_t crate_num, wib_num;
+} uvx_data;
```

Comment:
Upon further review, these structs really concern me...if they're supposed to be "flat data", then std::vector won't work. If they're supposed to be more like C++ structs, then I think a better data structure than 2-D C arrays of std::vectors should be used.

Reply:
The code in this file is a copy of Ben's WIB2 packing/unpacking code. I just included it in `dataformats` so that I can test the `WIB2Frame` "overlay"-style class against an independently-written implementation. (It's not really a great approach, but there aren't any instances of real WIB2 frames existing that I could test against, so this seemed the next best thing).

Since this code is just here for testing purposes, maybe I should just move it elsewhere. I could try putting it in `test/src`, if that gets picked up correctly by the build system, or maybe just move the code into the `WIB2Frame` unit test?